packages/sdk/src/chains/evm.ts

```diff
@@ -515,29 +519,25 @@ export class EvmChain implements IChain {
 		return gas
 	}
 
-	async estimateFillOrderGas(
-		order: Order,
-		venues: {
-			v2Router: HexString
-			v2Factory: HexString
-			v3Factory: HexString
-			v3Quoter: HexString
-		},
-		fillerWalletAddress: HexString,
-		intentGatewayAddress: HexString,
-		availableAssets: {
-			name: string
-			address: HexString
-			decimals: number
-		}[],
-		universalRouterAddress: HexString,
-		postGasEstimate: bigint,
-		wrappedNativeTokenSourceChain: HexString,
-	): Promise<bigint> {
-		const nativeTokenPriceUsd = await fetchTokenUsdPrice(wrappedNativeTokenSourceChain)
-		const gasPrice = await this.publicClient.getGasPrice()
+	async estimateFillOrderGas(order: Order, fillerWalletAddress: HexString, sourceChain: EvmChain): Promise<bigint> {
+		const postRequest: IPostRequest = {
+			source: order.destChain, // Destination Chain
+			dest: order.sourceChain, // Source Chain
+			body: constructRedeemEscrowRequestBody(order, fillerWalletAddress),
+			timeoutTimestamp: 0n,
+			nonce: await sourceChain.getHostNonce(),
```

Comment:
lets treat this `EvmChain` as the source & the provided client as the destination

Reply:
Sounds good.